lib/internal/streams/operators.js

```diff
@@ -150,6 +151,37 @@ async function * map(fn, options) {
   }
 }
 
+async function some(fn, options) {
+  // https://tc39.es/proposal-iterator-helpers/#sec-iteratorprototype.some
+  // Note that some does short circuit but also closes the iterator if it does
+  const ac = new AbortController();
+  if (options?.signal) {
+    options.signal.addEventListener('abort', () => ac.abort(), {
```

Comment:
What if `options.signal` is already aborted?

Reply:
Not related to this PR move it to the some/every PR?